litho-core/src/main/java/com/facebook/litho/ComponentTree.java

```diff
@@ -2143,7 +2163,15 @@ public LayoutState get() {
         }
       }
 
-      return layoutState.acquireRef();
+      if (isReleased()) {
+        // return early if our layout has been released already
+        return null;
+      }
+      try {
+        return layoutState.acquireRef();
```

Comment:
This doesn't seem threadsafe as isReleased() could return false but then before we acquire the ref, we release.

Reply:
It's not, the isReleased() check is also just a small best effort optimization to avoid creating the exception. I can remove that if you don't think it's a worthwhile optimization